toss/benchmarks/archipelago_benchmark.py

```diff
@@ -0,0 +1,175 @@
+    small_run_strong_scaling = np.empty((8,len(islands)), dtype=np.float64) # (8xlen(islands)) array with each column representing: [f, x, n_islands] (Here: f:1x1,  x:6x1, t:1x1)
+    for i in range(0,len(islands)):
+        number_of_islands = islands[i]
+        number_of_workers = number_of_islands
+        population_size = populations[i]
+        f_champion, x_champion, elapsed_time = find_solution(number_of_generations, number_of_islands, population_size)
+        small_run_strong_scaling[0,i] = f_champion
+        small_run_strong_scaling[1:7,i] = x_champion
+        small_run_strong_scaling[7,i] = elapsed_time
+
+        print("Elapsed time: ", elapsed_time)
+
+    np.savetxt("small_run_strong_scaling.csv", small_run_strong_scaling, delimiter=",")
+
+
+def strong_scale_large():
+    islands = [1, 2, 4, 8, 16, 32] 
+    populations = [320, 160, 80, 40, 20, 10]
+
+    #   large run:
+    print("Now initializing strong scaling: large run")
+    number_of_generations = 32
+    large_run_strong_scaling = np.empty((8,len(islands)), dtype=np.float64) # (8xlen(islands)) array with each column representing: [f, x, n_islands] (Here: f:1x1,  x:6x1, t:1x1)
+    for i in range(0,len(islands)):
+        number_of_islands = islands[i]
+        number_of_workers = number_of_islands
+        population_size = populations[i]
+        f_champion, x_champion, elapsed_time = find_solution(number_of_generations, number_of_islands, population_size)
+        large_run_strong_scaling[0,i] = f_champion
+        large_run_strong_scaling[1:7,i] = x_champion
+        large_run_strong_scaling[7,i] = elapsed_time
+
+        print("Elapsed time: ", elapsed_time)
+
+    np.savetxt("large_run_strong_scaling.csv", large_run_strong_scaling, delimiter=",")
+
+
+def weak_scale():
+
+    ######### Weak scaling: #########
+    print("Now initializing weak scaling run.")
+    # Scaling populations per thread (island) equivalent
+    #  to the number threads used to see if performance
+    #  is constants.
+    islands =  [1, 2, 4, 8, 16, 32]
+    number_of_generations = 32
+    populations = [10, 10, 10, 10, 10, 10]
+
+    weak_scaling = np.empty((8,len(islands)), dtype=np.float64) # (8xlen(islands)) array with each column representing: [f, x, n_islands] (Here: f:1x1,  x:6x1, t:1x1)
+    for i in range(0,len(islands)):
+        number_of_islands = islands[i]
+        number_of_workers = number_of_islands
+        population_size = populations[i]
+        f_champion, x_champion, elapsed_time = find_solution(number_of_generations, number_of_islands, population_size)
+        weak_scaling[0,i] = f_champion
+        weak_scaling[1:7,i] = x_champion
+        weak_scaling[7,i] = elapsed_time
+
+        print("Elapsed time: ", elapsed_time)
+
+    np.savetxt("weak_scaling.csv", weak_scaling, delimiter=",")
+
+
+def run():
+
+    # Initializing tests:
+    first_test()
+
+    #strong_scale_small()
+
+    #strong_scale_large()
+
+    #weak_scale()
+
+
+if __name__ == "__main__":
```

Comment:
Structure wise it may be nicer to have a top-level folder called scripts where you put things like this. It doesn't add any functionality to the module itself, so I am not sure it should be inside it?